### PR TITLE
nrf_desktop: Battery measurement on keyboard

### PR DESCRIPTION
**Pull request by Qbicz**
* Split battery charging from measurement
* Support measurement on keyboard, using VDD pin for ADC measurement

### CODE REVIEW

**Comment by Qbicz** [comment]
Tested on PCA20041 (with voltage divider and charger) and on PCA20037 (without both).

@pdunaj @MarekPieta can you review? @jhn-nordic you can take a look if you have time.

**Comment by Qbicz** [comment]
For the curious, I assumed sigmoid characteristic of AAA battery state of charge and generated it with
```
import numpy as np

def sigmoid(x):
    return (1 / (1 + np.exp(-x)))

si = np.linspace(-6, 6, 121)
batt_lvl = [int(100*x) for x in map(sigmoid, si)]

```

File: applications/nrf_desktop/src/hw_interface/Kconfig [hunk 1]
```diff
@@ -147,95 +147,118 @@ comment "Battery charger configuration"
...
 	int "Maximum battery voltage [mV] that corresponds to 100% level"
-	depends on DESKTOP_BATTERY_DISCRETE
+	depends on DESKTOP_BATTERY_MEAS
 	default 4180
 
-config DESKTOP_BATTERY_VOLTAGE_DIVIDER_UPPER
+config DESKTOP_BATTERY_MEAS_HAS_VOLTAGE_DIVIDER
+	bool "Use voltage divider for battery measurement"
+	depends on DESKTOP_BATTERY_MEAS
+	default n
```
Comment (pdunaj) [inline applications/nrf_desktop/src/hw_interface/Kconfig]: this is not needed

File: applications/nrf_desktop/configuration/nrf52_pca20037/app_ZDebug.conf [hunk 2]
```diff
@@ -19,6 +19,13 @@ CONFIG_DESKTOP_LED_ENABLE=y
 CONFIG_DESKTOP_LED_COUNT=1
 CONFIG_DESKTOP_LED_COLOR_COUNT=1
 
+CONFIG_DESKTOP_BATTERY_MEAS=y
+CONFIG_DESKTOP_BATTERY_MEAS_HAS_ENABLE_PIN=n
+CONFIG_DESKTOP_BATTERY_MEAS_MIN_LEVEL=2000
+CONFIG_DESKTOP_BATTERY_MEAS_MAX_LEVEL=3200
+CONFIG_DESKTOP_BATTERY_MEAS_HAS_VOLTAGE_DIVIDER=n
```
Comment (pdunaj) [inline applications/nrf_desktop/configuration/nrf52_pca20037/app_ZDebug.conf]: this value is n by default

File: applications/nrf_desktop/src/hw_interface/Kconfig [hunk 3]
```diff
@@ -147,95 +147,118 @@ comment "Battery charger configuration"
...
 
-config DESKTOP_BATTERY_CHARGE_ENABLE_PIN
-	int "Battery charging enable pin"
-	depends on DESKTOP_BATTERY_DISCRETE
-	help
-	  A pin which enables battery charging.
+config DESKTOP_BATTERY_MEAS_HAS_ENABLE_PIN
+	bool "Use pin for enabling battery measurement"
+	depends on DESKTOP_BATTERY_MEAS
+	default n
```
Comment (pdunaj) [inline applications/nrf_desktop/src/hw_interface/Kconfig]: default n is not needed

File: applications/nrf_desktop/src/hw_interface/Kconfig [hunk 4]
```diff
@@ -147,95 +147,118 @@ comment "Battery charger configuration"
...
 	prompt "Select battery charging interface"
-	default DESKTOP_BATTERY_NONE
+	default DESKTOP_BATTERY_CHARGER_NONE
 
-config DESKTOP_BATTERY_NONE
+config DESKTOP_BATTERY_CHARGER_NONE
 	bool "No battery charging"
 
-config DESKTOP_BATTERY_DISCRETE
+config DESKTOP_BATTERY_CHARGER
```
Comment (pdunaj) [inline applications/nrf_desktop/src/hw_interface/Kconfig]: I think I would stick with CHARGER_DISCRETE

Comment (Qbicz) [reply applications/nrf_desktop/src/hw_interface/Kconfig]: done

File: applications/nrf_desktop/src/hw_interface/Kconfig (hunk 3 above)
Comment (Qbicz) [reply applications/nrf_desktop/src/hw_interface/Kconfig]: done

File: applications/nrf_desktop/configuration/nrf52_pca20037/app_ZDebug.conf (hunk 2 above)
Comment (Qbicz) [reply applications/nrf_desktop/configuration/nrf52_pca20037/app_ZDebug.conf]: done

File: applications/nrf_desktop/src/hw_interface/Kconfig (hunk 1 above)
Comment (Qbicz) [reply applications/nrf_desktop/src/hw_interface/Kconfig]: done

**Comment by Qbicz** [comment]
@pdunaj done, also moved values for resistors from Kconfig to .conf.

**Comment by Qbicz** [comment]
CI passed. @pdunaj can you revisit?

**Pull request merged**